code/game/machinery/computer/robot.dm

```diff
@@ -59,6 +59,7 @@
 			module = R.module ? "[R.module.name] Module" : "No Module Detected",
 			synchronization = R.connected_ai,
 			emagged =  R.emagged,
+			can_user_act = R.connected_ai == user || !isAI(user),
```

Comment:
This is not checked in ui_act.
Since ui_act comes directly from clients, the values inputted can be anything and shouldn't be considered to be within the restrictions set in ui_data, or the ui code itself as clients can modify that by changing the byond cache.